crates/teloxide-core/src/types/rgb.rs

```diff
@@ -0,0 +1,96 @@
+use rgb::RGB8;
+use serde::{de::Visitor, Deserialize, Serialize};
+
+/// RGB color format
+#[derive(Clone, Copy, Debug)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Rgb {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+}
+
+impl Rgb {
+    fn to_u32(self) -> u32 {
+        u32::from_be_bytes([0, self.r, self.g, self.b])
+    }
+
+    fn from_u32(rgb: u32) -> Self {
+        let [_, r, g, b] = rgb.to_be_bytes();
+        Rgb { r, g, b }
+    }
+}
+
+impl Serialize for Rgb {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u32(self.to_u32())
+    }
+}
+
+impl<'de> Deserialize<'de> for Rgb {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct V;
+
+        impl Visitor<'_> for V {
+            type Value = Rgb;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("an integer represeting an RGB color")
+            }
+
+            fn visit_u32<E>(self, v: u32) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(Self::Value::from_u32(v))
+            }
+
+            fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_u32(v.try_into().map_err(|_| E::custom("rgb value doesn't fit u32"))?)
+            }
+        }
+
+        deserializer.deserialize_u32(V)
+    }
+}
+
+impl From<RGB8> for Rgb {
```

Comment:
I wanted to use `rgb::RGB8` directly, because it's very popular crate for work with RGB types. But it doesn't support telegram-way of rgb serialization, so I decided to make `From` conversion for users convenience.